ce_expansion/atomgraph/bcm.py

```diff
@@ -0,0 +1,207 @@
+
+            [precomps] = [gamma of element 1] * [ce_bulk of element 1 to element 2]
+
+        Returns: Precomp Matrix
+
+        """
+        # precompute values for BCM calc
+        n_met = len(self.metal_types)
+
+        precomps = np.ones((n_met, n_met))
+
+        for i in range(n_met):
+            for j in range(n_met):
+
+                M1 = self.metal_types[i]
+                M2 = self.metal_types[j]
+                precomp_bulk = self.ce_bulk[M1]
+                precomp_gamma = self.gammas[M1][M2]
+
+                precomps[i, j] = precomp_gamma * precomp_bulk
+        return precomps
+
+    def calc_ce(self, orderings):
+        """
+        Calculates the Cohesive energy of the ordering given or of the default ordering of the NP
+
+        [Cohesive Energy] = ( [precomp values of element A and B] / sqrt(12 * CN) ) / [num atoms]
+
+        Args:
+            orderings: The ordering of atoms within the NP; ordering key is based on Metals in alphabetical order
+                - Will use ardering defined by atom if not given an ordering
+
+        Returns: Cohesive Energy
+
+        """
+
+        # determine CN Values
+        (atom_CN, bond_CN) = np.hsplit(self.bond_list, 2)
+        (CN_Vals, CN) = np.unique(bond_CN, return_counts=True)
+
+        a1 = self.bond_list[:, 0]
+        a2 = self.bond_list[:, 1]
+
+        # creating bond orderings
+        return (self.precomps[orderings[a1], orderings[a2]] / np.sqrt(12 * CN[a1])).sum() / len(self.atoms)
+
+    def calc_ee(self, orderings=None):
+        """
+            Calculates the Excess energy of the ordering given or of the default ordering of the NP
+
+            [Excess Energy] = [CE of NP] - sum([Pure Element NP] * [Comp of Element in NP])
+
+        Args:
+            orderings: The ordering of atoms within the NP; ordering key is based on Metals in alphabetical order
+                - Will use ardering defined by atom if not given an ordering
+
+        Returns: Excess Energy
+
+        """
+
+        # If orderings = none, ensure tested atoms object contains desired orderings to test
+        if orderings is None:
+            orderings = [test_of_bcm.metal_types.index(x) for x in atoms.symbols]
+
+        metals = np.bincount(orderings)
+
+        #Obtain atom fractions of each tested element
+        x_i =  metals / len(orderings)
+
+        #Calculate energy of tested NP first;
+        ee = self.calc_ce(orderings)
+
+       # Then, subtract calculated pure NP energies multiplied by respective fractions to get Excess Energy
+        for ele in range(len(metals)):
+            x_ele = x_i[ele]
+            o_mono_x = np.ones(len(self), int) * ele
+
+            ee -= self.calc_ce(o_mono_x) * x_ele
+        return(ee)
+
+    def calc_smix(self, orderings=None):
+        """
+        compositions = np.bincount(ordering) / len(ordering)
+
+        kb = ase.units.Kb
+        smix = -kb * sum(xi * ln(xi) for xi in compositions)
+        sum(compositions) == 1
```

Comment:
Rework this doc string to remove unnecessary code snippets. You can also add `Args:` and `Returns:` sections.